src/main/java/gregtech/common/covers/filter/BaseFilterContainer.java

```diff
@@ -201,9 +200,15 @@ public void deserializeNBT(NBTTagCompound nbt) {
     }
 
     public void handleLegacyNBT(NBTTagCompound nbt) {
-        if (hasFilter()) {
-            getFilter().getFilterReader().handleLegacyNBT(nbt);
+        // for filters as covers, the stack is set manually, and "FilterInventory" doesn't exist to be deserialized
+        // also, ItemStackHandler's deserialization doesn't use setStackInSlot, so I have to do that manually here
+        if (nbt.hasKey("FilterInventory")) {
```

Comment:
Same question here.

Reply:
"FilterInventory" only exists when the cover used a FilterContainer (pumps, conveyors, etc). Filters as a cover only used the FilterWrapper, which does not have "FilterInventory" written to the tag. with the filter rework, "FilterInventory" is always written.